### PR TITLE
Create slowsort.py

### PR DESCRIPTION
**Pull request by MrGeek1337**
added slowsort algorithm implementation to sorts

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by mrmaxguns** [comment]
I believe this algorithm has been already added [here](https://github.com/TheAlgorithms/Python/blob/7f04e5cd3499c07f07ae94437b706254edb7ba39/sorts/bogo_sort.py).

**Comment by MrGeek1337** [comment]
The algorithm you linked to is Bogosort, which is an entirely different algorithm, but it is also called Slowsort at times.

**Comment by mrmaxguns** [comment]
Oh, ok. Sorry for the misunderstanding

**Comment by MrGeek1337** [comment]
No worries, you could look at the code and notice the differences, you could also check [https://en.wikipedia.org/wiki/Bogosort](url) and [https://en.wikipedia.org/wiki/Slowsort](url) at Wikipedia.

File: sorts/slowsort.py
Comment (mrmaxguns) [inline sorts/slowsort.py:15]: Python's `typing` module has a way of dealing with arguments that default to None. Also, a should be lowercase, as by python convention:

```python
from typing import Optional

def slowsort(a: list, i: Optional[int] = None, j: Optional[int] = None) -> None:
    ...
```

Comment (MrGeek1337) [reply sorts/slowsort.py:15]: I first thought it was too much to use `Typing`, but I guess you're right in including it, so I'll include it, Thanks.

**Comment by mrmaxguns** [comment]
Isort failed, looks like you need to reorder the imports

**Comment by MrGeek1337** [comment]
Done!

**Comment by mrmaxguns** [comment]
Thanks for contributing to open source :thumbsup:

@cclauss could you review and merge this pull request?

File: sorts/slowsort.py
Comment (NumberPiOso) [inline sorts/slowsort.py:48]: `m = (i + j) // 2`

I think it is a bit cleaner and prevent you from having to import math.floor

**Review by NumberPiOso — CHANGES_REQUESTED** [review]

File: sorts/slowsort.py
Comment (MrGeek1337) [reply sorts/slowsort.py:48]: Yes indeed, thanks.

**Comment by xcodz-dot** [comment]
@dhruvmanila, check this out

File: sorts/slowsort.py
Comment (xcodz-dot) [inline sorts/slowsort.py:15]: Give (i, j) a descriptive name please

File: sorts/slowsort.py
Comment (xcodz-dot) [inline sorts/slowsort.py:38]:
```suggestion
    j defaults to len(sequence) - 1 if not given.
    It returns None.
    
    >>> sequence = [1, 6, 2, 5, 3, 4, 4, 5]
    >>> slowsort(sequence)
    >>> sequence
    [1, 2, 3, 4, 4, 5, 5, 6]
    
    >>> sequence = []
    >>> slowsort(sequence)
    >>> sequence
    []
    
    >>> sequence = [2]
    >>> slowsort(sequence)
    >>> sequence
    [2]
    
    >>> sequence = [1, 2, 3, 4]
    >>> slowsort(sequence)
    >>> sequence
    [1, 2, 3, 4]
    
    >>> sequence = [4, 3, 2, 1]
    >>> slowsort(sequence)
    >>> sequence
    [1, 2, 3, 4]
    
    >>> sequence = [9, 8, 7, 6, 5, 4, 3, 2, 1, 0]
    >>> slowsort(sequence, 2, 7)
    >>> sequence
    [9, 8, 2, 3, 4, 5, 6, 7, 1, 0]
    
    >>> sequence = [9, 8, 7, 6, 5, 4, 3, 2, 1, 0]
    >>> slowsort(sequence, j = 4)
    >>> sequence
    [5, 6, 7, 8, 9, 4, 3, 2, 1, 0]
    
    >>> sequence = [9, 8, 7, 6, 5, 4, 3, 2, 1, 0]
    >>> slowsort(sequence, i = 5)
    >>> sequence
    [9, 8, 7, 6, 5, 0, 1, 2, 3, 4]
    """
    if i is None:
```

Semicolons are not good to be used

**Review by xcodz-dot — CHANGES_REQUESTED** [review]

File: sorts/slowsort.py
Comment (cclauss) [reply sorts/slowsort.py:38]: Why add all these blank lines?  Tests are not more important than the code.

**Comment by dhruvmanila** [comment]
@algorithms-keeper review

**Comment by mrmaxguns** [comment]
A maintainer should probably merge this PR unless there's an issue; we don't want valid pull requests to be closed as stale :disappointed:

**Comment by cclauss** [comment]
@mrmaxguns If you believe that this PR is ready to be landed, please click the `Review changes` button at https://github.com/TheAlgorithms/Python/pull/3865/files and Approve.

**Review by mrmaxguns — APPROVED** [review]
This pull request is good to good to go:
- [X] Unique
- [X] Docstrings
- [X] Type hints
- [X] Readability
- [X] Doctests

**Review by xcodz-dot — APPROVED** [review]
this looks good to go, although i would like to ask if it is prone to recursion error

**Comment by cclauss** [comment]
@mateuszz0000 Is this ready to land?

**Comment by MrGeek1337** [comment]
> 
> 
> this looks good to go, although i would like to ask if it is prone to recursion error

Nothing but a `maximum recursion depth exceeded` with large lists, but this is the case with all recursive functions (in Python), and this algorithm isn't really made for practical use, so it isn't a necessary thing to use `sys.setrecursionlimit` to try and overcome the error.

**Review by cclauss — APPROVED** [review]
I am not a fan of purposefully inefficient algorithms of little utility (esp. sleep_sort!) -- perhaps the _humourous nature_ is beyond my comprehension.  Heating the planet with inefficient algorithms is what TheAlgorithms are trying to avoid.  But this one has been batted about long enough so let's land it...

**Pull request merged**
